OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitYoutubeVideoFragment.java

```diff
@@ -0,0 +1,450 @@
+                            break;
+                        }
+                        currentSubtitleIndex++;
+                    }
+                }
+            }
+            subtitleDisplayHandler.postDelayed(this, SUBTITLES_DISPLAY_DELAY_MS);
+        }
+    };
+
+
+    private void initializeYoutubePlayerFragment() {
+
+        try {
+            if (getUserVisibleHint() && youTubePlayerFragment != null && NetworkUtil.verifyDownloadPossible((BaseFragmentActivity) getActivity())) {
+
+                initTranscripts();
+                String apiKey = environment.getConfig().getEmbeddedYoutubeConfig().getYoutubeApiKey();
+                if (apiKey == null || apiKey.isEmpty()) {
+                    return;
+                }
+                youTubePlayerFragment.initialize(apiKey, this);
+            }
+        }
+        catch (NullPointerException e) {
+            return;
+        }
+    }
+
+    private void loadTranscriptsData(VideoBlockModel unit) {
+        TranscriptManager transcriptManager = new TranscriptManager(getContext());
+        TranscriptModel transcript = getTranscriptModel(unit.getDownloadEntry(environment.getStorage()));
+        transcriptManager.downloadTranscriptsForVideo(transcript);
+
+        try {
+            LinkedHashMap<String, InputStream> localHashMap = transcriptManager
+                    .fetchTranscriptsForVideo(transcript);
+            if (localHashMap != null) {
+                for (String thisKey : localHashMap.keySet()) {
+                    InputStream localInputStream = localHashMap.get(thisKey);
+                    if (localInputStream != null) {
+                        TimedTextObject localTimedTextObject =
+                                new FormatSRT().parseFile("temp.srt", localInputStream);
+                        srtList.put(thisKey, localTimedTextObject);
+                        localInputStream.close();
+                    }
+                }
+                if (!srtList.entrySet().isEmpty()) {
+                    setSubtitlesObj();
+                }
+            }
+
+        } catch (Exception localException) {
+            logger.error(localException);
+        }
+    }
+
+    private void setSubtitlesObj(){
+        String key = Locale.getDefault().getLanguage();
+        if (key.equals("iw")) {
+            key = "he";
+        }
+        if (srtList.containsKey(key)) {
+            subtitlesObj = srtList.get(key);
+        } else if (srtList.containsKey("en")) {
+            subtitlesObj = srtList.get("en");
+        } else {
+            subtitlesObj = srtList.entrySet().iterator().next().getValue();
+        }
+
+    }
+
+    private boolean initTranscripts(){
+        loadTranscriptsData(unit);
+        try {
+            initTranscriptListView();
+            updateTranscript(subtitlesObj);
+            return true;
+        }
+        catch (NullPointerException e) {
+            return false;
+        }
+    }
+
+    private void removeFragment(){
```

Comment:
method should be update accordingly after updating the code for YouTube Player fragment and YouTube player initializing.